src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,103 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    private int hash(K key) {
+        int index = key.hashCode() & (table.length - 1);
+        return (index == 0) ? 1 : index;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] copyTable = table;
+        table = new Node[table.length * 2];
+        for (Node<K, V> node : copyTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        if (key == null) {
+            putKeyNull(key, value);
+            return;
+        }
+        int index = hash(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else if (table[index].key.equals(key)) {
+            table[index].value = value;
+            return;
+        }
+        Node<K, V> lastNode = table[index];
+        while (lastNode.next != null) {
+            lastNode = lastNode.next;
+        }
+        lastNode.next = new Node<>(key, value, null);
+        size++;
+    }
 
+    private void putKeyNull(K key, V value) {
+        if (table[0] == null) {
+            table[0] = new Node<>(key, value, null);
+            size++;
+        } else {
+            table[0].value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
```

Comment:
let's simplify this method 
![image](https://user-images.githubusercontent.com/36365171/71007391-99555380-20ef-11ea-9785-0031e25bd453.png)

Reply:
Thank you very much. It looks now much better!!